python/pyspark/sql/dataframe.pyi

```diff
@@ -86,7 +86,7 @@ class DataFrame(PandasMapOpsMixin, PandasConversionMixin):
     def withWatermark(
         self, eventTime: ColumnOrName, delayThreshold: str
     ) -> DataFrame: ...
-    def hint(self, name: str, *parameters: Any) -> DataFrame: ...
+    def hint(self, name: str, *parameters: Union[str, list, float, int]) -> DataFrame: ...
```

Comment:
I think it is better to stick to `typing.List` with type bound (even if it's going to be `Any` instead of list here).

Reply:
Hm, let me just stick to `Any` for simplicity for now - actually this should be `Union[List[Union[str, list, float, int], str, list, float, int]`, right.

Reply:
We have `pyspark._typing.PrimitiveType` which covers things, that can have simple 1:1 mapping to JVM (all above + `bool`), so something around these lines 

```python
def hint(self, name: str, *parameters: Union[PrimitiveType, List[PrimitiveType]]) -> DataFrame: ...
```

might work.